index.js

```diff
@@ -861,8 +861,16 @@ RedisClient.prototype.end = function () {
 function Multi(client, args) {
     this._client = client;
     this.queue = [["multi"]];
+    var command, tmp_args;
     if (Array.isArray(args)) {
-        this.queue = this.queue.concat(args);
+        while (tmp_args = args.shift()) {
+            command = tmp_args.shift();
+            if (Array.isArray(command)) {
+                this[command[0]].apply(this, command.slice(1).concat(tmp_args));
+            } else {
+                this[command].apply(this, tmp_args);
```

Comment:
To find a consistent way to handle the commands everywhere the same I had to use this slightly more complex logic. But since this is a very special case anyway this should not matter.